src/main/java/edu/kit/datamanager/repo/util/ValidatorUtil.java

```diff
@@ -0,0 +1,118 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.kit.datamanager.repo.util;
+
+import edu.kit.datamanager.exceptions.UnsupportedMediaTypeException;
+import edu.kit.datamanager.repo.util.validators.IIdentifierValidator;
+import edu.kit.datamanager.repo.util.validators.impl.DOIValidator;
+import edu.kit.datamanager.repo.util.validators.impl.HandleNetValidator;
+import edu.kit.datamanager.repo.util.validators.impl.ISBNValidator;
+import edu.kit.datamanager.repo.util.validators.impl.URLValidator;
+import org.datacite.schema.kernel_4.RelatedIdentifierType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * @author maximilianiKIT
+ */
+public class ValidatorUtil {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ValidatorUtil.class);
+    private static final ValidatorUtil soleInstance = new ValidatorUtil();
+
+    private static final Set<IIdentifierValidator> validators;
+
+    static {
+        Set<IIdentifierValidator> validators1 = new HashSet<>();
+
+        validators1.add(new HandleNetValidator());
+        validators1.add(new DOIValidator());
+        validators1.add(new URLValidator());
+        validators1.add(new ISBNValidator());
+
+        validators = validators1;
+    }
+
+    /**
+     * This private constructor enforces singularity.
+     */
+    private ValidatorUtil() {
+    }
+
+    /**
+     * This method returns the singleton.
+     *
+     * @return singleton instance of this class
+     */
+    public static ValidatorUtil getSingleton() {
+        return soleInstance;
+    }
+
+    /**
+     * This method returns a list of the types of all implemented validators.
+     *
+     * @return a list of RelatedIdentifierType.
+     */
+    public List<RelatedIdentifierType> getAllAvailableValidatorTypes() {
+        LOGGER.debug("getAllAvailableValidatorTypes");
+        List<RelatedIdentifierType> result = new ArrayList<>();
+        for (IIdentifierValidator i : validators) result.add(i.getSupportedType());
+        LOGGER.info("All available validator types: {}", result.toString());
+        return result;
+    }
+
+    /**
+     * This method checks if the type passed in the parameter is valid and then uses the corresponding validator to check the input.
+     *
+     * @param input The input which gets validated.
+     * @param type  The type of the validator.
+     * @return true if the input and type are valid.
+     * May throw an exception if the input or the type is invalid or other errors occur.
+     */
+    public boolean isValid(String input, RelatedIdentifierType type) {
+        LOGGER.debug("isValid - string type");
+        for (IIdentifierValidator i : validators) {
```

Comment:
I disagree, because:
1. There are only 18 available validator types.
2. This set ensures that each validator is unique.

I agree, because:
1. A map is faster.
2. A map is easier to access.

What do you think?

Reply:
Why should a set ensure that there is only one unique validator? There might be several validators implementing URL which will cause problems (which one is the correct one). There might be also several validators of the same class as the instances of a validator class is not a singleton.  Furthermore you have no clue in which order the validators appears.
For HashMap<type, validator>  there is only one validator for each type! A Map costs not more resources than a Set. (Ok,  there is a really small overhead which allows you to access the element directly)
--> No reason to use Set instead of a Map.